components/concept/sync/src/main/java/mozilla/components/concept/sync/OAuthAccount.kt

```diff
@@ -169,9 +172,28 @@ interface OAuthAccount : AutoCloseable {
      * @param sessionToken token string to use for login
      * @param kSync sync string for login
      * @param kXCS XCS string for login
-     * @return Deferred boolean success or failure for the migration event
+     * @return JSON object with the result of the migration or 'null' if it failed.
+     * For up-to-date schema, see underlying implementation in https://github.com/mozilla/application-services/blob/v0.49.0/components/fxa-client/src/migrator.rs#L10
+     * At the moment, it's just "{total_duration: long}".
      */
-    fun copyFromSessionTokenAsync(sessionToken: String, kSync: String, kXCS: String): Deferred<Boolean>
+    fun copyFromSessionTokenAsync(sessionToken: String, kSync: String, kXCS: String): Deferred<JSONObject?>
+
+    /**
+     * Checks if there's a migration in-flight. An in-flight migration means that we've tried to migrate
+     * via either [migrateFromSessionTokenAsync] or [copyFromSessionTokenAsync], and failed for intermittent
+     * (e.g. network)
+     * reasons. When an in-flight migration is present, we can retry using [retryMigrateFromSessionTokenAsync].
+     * @return Boolean 'true' if there's an in-flight migration state present.
+     */
+    fun isInMigrationState(): Boolean
```

Comment:
Over in https://github.com/mozilla/application-services/pull/2586 I'm introducing a new `MigrationState` enum to use instead of a boolean here, but am happy to take advice on an alternative approach if there's one that would fit better here.